docs/sql-ref-literals.md

```diff
@@ -21,14 +21,74 @@ license: |
 
 A literal (also known as a constant) represents a fixed data value. Spark SQL supports the following literals:
 
+ * [Typed Literal](#typed-literal)
  * [String Literal](#string-literal)
  * [Binary Literal](#binary-literal)
  * [Null Literal](#null-literal)
  * [Boolean Literal](#boolean-literal)
  * [Numeric Literal](#numeric-literal)
+ * [Timestamp Literal](#timestamp-literal)
  * [Datetime Literal](#datetime-literal)
  * [Interval Literal](#interval-literal)
 
+### Typed Literal
```

Comment:
I feel "Typed Literal" is a bit confusing. For example, a numerical literal is also a typed literal, right? How about saying `Typed Literal for Binary, Date, Timestamp, and Interaval` instead?

Reply:
> I feel "Typed Literal" is a bit confusing. For example, a numerical literal is also a typed literal, right? How about saying `Typed Literal for Binary, Date, Timestamp, and Interaval` instead?

LGTM, changed